### PR TITLE
Adding new templates for google-native

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

**Review by infin8x — APPROVED** [review]
These look good! (And I've confirmed that the blog post sample code for `pulumi new` in each language matches your template names)

File: google-native-csharp/${PROJECT}.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Pulumi.GoogleNative" Version="0.0.*-alpha*" />
```
Comment (mikhailshilkov) [inline google-native-csharp/${PROJECT}.csproj:10]:
```suggestion
    <PackageReference Include="Pulumi.GoogleNative" Version="0.*" />
```
I assume we'll publish 0.1.0 before merging

File: google-native-go/go.mod [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module ${PROJECT}
+
+go 1.16
+
+require (
+	github.com/pulumi/pulumi-google-native/sdk v0.0.0-20210416221815-46b60b89ac41
+	github.com/pulumi/pulumi/sdk/v3 v3.0.0-rc.1
```
Comment (mikhailshilkov) [inline google-native-go/go.mod:7]: Reminder to update this before merging

**Review by mikhailshilkov — APPROVED** [review]
LGTM modulo updating all reference after the release

**Comment by viveklak** [comment]
> LGTM modulo updating all references after the release

Sounds good. Will hold till updated versions are released, then update.

File: google-native-csharp/${PROJECT}.csproj (hunk 1 above)
Comment (viveklak) [reply google-native-csharp/${PROJECT}.csproj]: Done.

**Pull request merged**